lib/internal/per_context/domexception.js

```diff
@@ -49,12 +49,25 @@ const disusedNamesSet = new SafeSet()
   .add('ValidationError');
 
 class DOMException {
-  constructor(message = '', name = 'Error') {
+  constructor(message = '', options = 'Error') {
     ErrorCaptureStackTrace(this);
-    internalsMap.set(this, {
-      message: `${message}`,
-      name: `${name}`
-    });
+
+    if (options && typeof options === 'object') {
+      const { name } = options;
+      internalsMap.set(this, {
+        message: `${message}`,
+        name: `${name}`
+      });
+
+      if ('cause' in options) {
+        ObjectDefineProperty(this, 'cause', { __proto__: null, value: cause });
```

Comment:
If we want to align with ECMAScript `cause`, it would need to be writable and enumerable (which I think is what the WebIDL is going for?)
```suggestion
        ObjectDefineProperty(this, 'cause', {
          __proto__: null,
          configurable: true,
          value: cause,
          writable: true,
        });
```

Reply:
By the definition of ecma262, `cause` is installed as a non-enumerable data property: https://tc39.es/ecma262/#sec-installerrorcause.

Reply:
I meant configurable, not enumerable, sorry.